src/archivist/services/index.js

```diff
@@ -110,14 +124,34 @@ export async function loadWithHistory(servicesIds = []) {
           });
         }
 
+        if (!combine) {
+          pages.push(new PageDeclaration({
+            location: declarationForThisDate.fetch,
+            executeClientScripts: declarationForThisDate.executeClientScripts,
+            contentSelectors: declarationForThisDate.select,
+            noiseSelectors: declarationForThisDate.remove,
+            filters: actualFilters,
+          }));
+        } else {
+          for (const pageDeclaration of combine) {
+            const { filter: pageFilterNames, fetch: pageLocation, pageExecuteClientScripts, select: pageContentSelectors, remove: pageNoiseSelectors } = pageDeclaration;
+
+            const pageFilters = await loadServiceFilters(serviceId, pageFilterNames); // eslint-disable-line no-await-in-loop
+
+            pages.push(new PageDeclaration({
+              location: pageLocation || declarationForThisDate.fetch,
+              executeClientScripts: pageExecuteClientScripts || declarationForThisDate.executeClientScripts,
+              contentSelectors: pageContentSelectors || declarationForThisDate.select,
+              noiseSelectors: pageNoiseSelectors || declarationForThisDate.remove,
+              filters: pageFilters || actualFilters,
```

Comment:
Extract to function and call it in both multi and single page documents to decrease complexity and risk of inconsistent behaviour 🙂 If possible, move it outside of this file to decrease its size (it's approaching 200 SLoC 😕).

> For example, add a factory static method `PageDeclaration.from` that instantiates a `PageDeclaration` by renaming `fetch` to `location` etc, and let it take an optional `defaults` object as second parameter, which would be empty for single page documents.